yarl/__init__.py

```diff
@@ -164,10 +164,17 @@ def __new__(cls, val='', *, encoded=False, strict=None):
                 if host is None:
                     raise ValueError(
                         "Invalid URL: host is required for abolute urls.")
+
+                try:
+                    port = val.port
+                except ValueError:
+                    raise ValueError(
+                        "Invalid URL: port can't be converted to integer")
```

Comment:
Maybe `raise ... from ...`?

Reply:
Implicit exception context is good enough